### PR TITLE
Gimbal joint limits

### PR DESCRIPTION
**Pull request by hsu**
Fix the problem where gimbal goes unstable when following commands are given:

```
make posix_sitl_default  gazebo_typhoon_h480 
commander takeoff
vmount test yaw 90
```

### CODE REVIEW

**Comment by hsu** [comment]
@bkueng

**Comment by LorenzMeier** [comment]
@bkueng Please merge once it passes your testing.

File: src/gazebo_gimbal_controller_plugin.cpp [hunk 1]
```diff
@@ -354,47 +365,99 @@ void GimbalControllerPlugin::OnUpdate()
...
+      pitchError = pitchError1;
+    }
+    double rollError1 = currentAnglePRYVariable.Y() - this->rollCommand;
     if (std::abs(rollError) > std::abs(rollError1))
     {
       rollError = rollError1;
     }
+    else
+    {
+      rollError = rollError1;
```
Comment (bkueng) [inline src/gazebo_gimbal_controller_plugin.cpp]: This is doing the same thing in the if & else clause

**Comment by bkueng** [comment]
Thanks @hsu, it works now. I had to raise the yaw joint limits to +-PI to get past 90deg.
However when I do:

```
vmount test yaw 180
vmount test yaw -170
```

It rotates almost the full circle instead of just doing the 10deg adjustment. Can you fix this please?

**Comment by hsu** [comment]
@bkueng 

I raised gimbal yaw limits as you suggested:

```
diff --git a/models/typhoon_h480/typhoon_h480.sdf b/models/typhoon_h480/typhoon_h480.sdf
index 9a85acf..61a97dd 100644
--- a/models/typhoon_h480/typhoon_h480.sdf
+++ b/models/typhoon_h480/typhoon_h480.sdf
@@ -243,8 +243,8 @@
       <axis>
         <xyz>0 0 1</xyz>
         <limit>
-          <lower>-1.5</lower>
-          <upper>1.5</upper>
+          <lower>-3.14159</lower>
+          <upper>3.14159</upper>
           <effort>100</effort>
           <velocity>-1</velocity>
         </limit>
```

and here's my test result for yaw near 180°:

### wrap around from 180 to -170

Given the joint is not a continuous joint, I don't think it can move past 180° to 190° (equivalent of -170°). Is the gimbal yaw a continuous joint on the h480?

### issue - gimbal stuck at +/-180°

```
vmount test yaw 180
```

gimbal moves to 180 correctly.

```
vmount test yaw 170
```

gimbal appears to be stuck at 180. I have to give several commands to get it "unstuck" randomly.

However, if I start by sending the gimbal to yaw of 179°, I can then move the gimbal freely without getting the "stuck" behavior.

I'll test some more to see why this is happening. I have a strong feeling there's still bug in the rotation math, suggestions welcome. Thanks!

**Comment by LorenzMeier** [comment]
@hsu The gimbal can indeed rotate freely. The original has a slip ring.

**Comment by hsu** [comment]
@LorenzMeier got it, will test continuous yaw control. While I am updating the model, what are the joint limits for roll and pitch?

**Comment by bkueng** [comment]
For the pitch, limits are [-90, 5] deg, for roll I don't know, I'd use +-45 or 90deg.

**Comment by hsu** [comment]
Wrap around problem and getting stuck at 180° should be fixed.

Joint limits updated.

p.s. I also made a mistake pushing to PX4/sitl_gazebo branch gimbal_joint_limits of the same fix, I can delete that branch since we are doing PR from the osrf fork.

**Comment by bkueng** [comment]
Thanks @hsu, this works fine now. I continued in #46

**Pull request closed without merging**